pkg/detectors/bulksms/bulksms.go

```diff
@@ -24,7 +25,7 @@ var (
 	client = common.SaneHttpClient()
 
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives
-	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"bulksms"}) + `\b([a-fA-Z0-9*]{29})\b`)
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"bulksms"}) + `\b([a-zA-Z0-9!@#$%^&*()]{29})\b`)
```

Comment:
I generated over 10 tokens, and it appears the new regex is correct. So far, I’ve only received tokens containing the special characters `!_*#`, though I believe other special characters are possible.

**Potential Improvements for the Detector:**

- Instead of manually encoding the key and ID for basic authentication, we can leverage the SetBasicAuth function to simplify the process.
- Currently, we consider the result verified if the API returns a status code between 200 and 300. We could improve this by being more specific. Let's add the comment to reference the API docs as well.
- Based on my observations, both the ID and key are unique for each token generation. If a key is verified successfully with any ID, we can skip the remaining verification attempts for that key and proceed to the next one.

Reply:
Thanks for the review @kashifkhan0771 , I have addressed the changes